vm/src/hint_processor/builtin_hint_processor/bootloader_hints.rs

```diff
@@ -272,4 +372,158 @@ mod tests {
+            },
+            packed_outputs: packed_outputs.clone(), 
+        };
+
+        let mut vm = vm!();
+        let mut exec_scopes = ExecutionScopes::new();
+
+        exec_scopes.insert_box("bootloader_input", Box::new(bootloader_input.clone()));
+
+        let hint_data =
+            HintProcessorData::new_default(String::from(BOOTLOADER_SAVE_PACKED_OUTPUTS), HashMap::new());
+        let mut hint_processor = BuiltinHintProcessor::new_empty();
+        assert_matches!(
+            hint_processor.execute_hint(
+                &mut vm,
+                &mut exec_scopes,
+                &any_box!(hint_data),
+                &HashMap::new(),
+            ),
+            Ok(())
+        );
+
+        let saved_packed_outputs = exec_scopes.get::<Vec<PackedOutput>>("packed_outputs");
+        assert_matches!(
+            saved_packed_outputs,
+            Ok(ref x) if x == &packed_outputs
+        );
+
+        assert_eq!(saved_packed_outputs.expect("asserted Ok above, qed").len(), 3);
+    }
+
+    #[test]
+    fn test_set_packed_output_to_subtasks_hint() {
+        use crate::hint_processor::builtin_hint_processor::hint_code::BOOTLOADER_SET_PACKED_OUTPUT_TO_SUBTASKS;
+        use felt::Felt252;
+
+        let mut vm = vm!();
+        let mut exec_scopes = ExecutionScopes::new();
+
+        exec_scopes.insert_box("packed_output", Box::new(Felt252::from(42)));
+
+        let hint_data =
+            HintProcessorData::new_default(String::from(BOOTLOADER_SET_PACKED_OUTPUT_TO_SUBTASKS), HashMap::new());
+        let mut hint_processor = BuiltinHintProcessor::new_empty();
+        assert_matches!(
+            hint_processor.execute_hint(
+                &mut vm,
+                &mut exec_scopes,
+                &any_box!(hint_data),
+                &HashMap::new(),
+            ),
+            Ok(())
+        );
+
+        let packed_outputs = exec_scopes.get::<Felt252>("packed_outputs");
+        assert_matches!(
+            packed_outputs,
+            Ok(x) if x == Felt252::from(42)
+        );
+    }
+
+    #[test]
+    fn test_guess_pre_image_of_subtasks_output_hash_hint() {
+        use crate::hint_processor::builtin_hint_processor::hint_code::BOOTLOADER_GUESS_PRE_IMAGE_OF_SUBTASKS_OUTPUT_HASH;
+
+        let mut vm = vm!();
+        let mut exec_scopes = ExecutionScopes::new();
+
+        exec_scopes.insert_box("packed_output", Box::new(PackedOutput {}));
+
+        let hint_data =
+            HintProcessorData::new_default(String::from(BOOTLOADER_GUESS_PRE_IMAGE_OF_SUBTASKS_OUTPUT_HASH), HashMap::new());
+        let hint_data = any_box!(hint_data);
+        let mut hint_processor = BuiltinHintProcessor::new_empty();
+        assert_matches!(
+            hint_processor.execute_hint(
+                &mut vm,
+                &mut exec_scopes,
+                &hint_data,
+                &HashMap::new(),
+            ),
+            Ok(())
+        );
```

Comment:
There's a test helper that simplifies this a lot.

```suggestion
            run_hint!(
                vm,
                ids_data,
                hint_code::BOOTLOADER_GUESS_PRE_IMAGE_OF_SUBTASKS_OUTPUT_HASH,
                &mut exec_scopes
            )
```

Reply:
This may still be useful in the other tests in this PR, I'll review that...

Reply:
You don't particularly need `hint_data`: You can just reuse `ids_data` directly and set `ap_tracking = ApTracking::new();`. Another solution is to proceed like I did in my tests and call the hint directly. The disadvantage is that you don't check if your hint is mapped correctly though.

Reply:
Fixed in 20804f2. I'm now duplicating `hint_data` (since it's also created in `run_hint!`) but that doesn't seem to be a problem.

Reply:
...further cleaned up in ac524e6